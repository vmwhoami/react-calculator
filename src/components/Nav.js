import React from 'react';
import { NavLink } from 'react-router-dom';

const Nav = () => (

  <div className="nav-container">

    <NavLink to="/">
      <h1>Math magicians</h1>
    </NavLink>
    <nav>
      <NavLink to="/">Home</NavLink>
      <NavLink to="/calculator">Calculator</NavLink>
      <NavLink to="/quote">Quote</NavLink>
    </nav>
  </div>
);

export default Nav;
