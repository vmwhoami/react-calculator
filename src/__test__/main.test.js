import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Nav from '../components/Nav';
import Home from '../pages/home';
import Calculator from '../pages/calculator';
import Quote from '../pages/quote';

it('main renders correctly', () => {
  const tree = renderer
    .create(
      <div className="container">
        <BrowserRouter>
          <Nav />
          <Switch>
            <Route path="/" component={Home} exact />
            <Route path="/calculator" component={Calculator} />
            <Route path="/quote" component={Quote} />
          </Switch>
        </BrowserRouter>
      </div>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

