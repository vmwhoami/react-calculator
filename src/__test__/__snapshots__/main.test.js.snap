// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="nav-container"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <h1>
        Math magicians
      </h1>
    </a>
    <nav>
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Home
      </a>
      <a
        aria-current={null}
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
      <a
        aria-current={null}
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </nav>
  </div>
  <main
    className="maincont"
  >
    <article>
      <h1>
        Welcome to our page!
      </h1>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Earum, consequatur deserunt sed temporibus optio illo dolorum aperiam, repellat ipsum est omnis dolores iure molestiae voluptas perferendis ex officia! Porro, deserunt.
      </p>
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Earum, consequatur deserunt sed temporibus optio illo dolorum aperiam, repellat ipsum est omnis dolores iure molestiae voluptas perferendis ex officia! Porro, deserunt.
      </p>
    </article>
  </main>
</div>
`;
